### PR TITLE
Fix #7352: Add tracking and ad block level setting

### LINKED ISSUE #7352: Add standard (non-aggressive) Adblock mode
### Description: 
We mostly use standard ad-blocking in iOS except for cosmetic filtering. The intention is to add a option to use standard or aggressive tracking and privacy blocking in iOS to conform to the same standards as desktop as seen in the following screenshots:

Screenshot 2023-04-27 at 11 56 32 AM

Screenshot 2023-04-27 at 11 56 53 AM

Here is what we currently have in iOS:
Simulator Screen Shot - iPhone 14 Pro - 2023-04-27 at 11 58 30
Screenshot 2023-04-27 at 11 58 20 AM

### PR DESCRIPTION
**Pull request by cuba**
## Summary of Changes

This pull request fixes #7352 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

1. Test any known cosmetic filtering for pages like reddit, Some known test are:
2. Go to brave search and search for "trezor". 
- The ad on the top of the page should be hidden on aggressive
- The ad on the top of the page should be shown on standard
3. Enable "Block cookie consent notices" and go to reddit. 
- Cookie notices should be blocked on standard and aggressive
5. Enable "Block open in app notices" and go to reddit.
- Open in app notices should be blocked on standard and aggressive
7. Go to the following page: https://dev-pages.brave.software/filtering/cosmetic-filtering.html
- On aggressive mode, all images should be hidden after clicking "Run tests"
- On aggressive mode, only 3rd party images should be hidden after clicking "Run tests" (there is an unhiding delay)

## Screenshots:

Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 16 53 59
Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 16 53 53
Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 16 53 43
Simulator Screenshot - iPhone 14 Pro - 2023-05-10 at 16 53 36

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/Preferences/GlobalPreferences.swift [hunk 1]
```diff
@@ -34,10 +34,82 @@ extension Preferences {
...
+      }
+      
+      /// Return the P3A answer associated with this enum
+      public var p3AAnswer: P3AAnswer {
+        switch self {
+        case .disabled: return .disabled
+        case .standard: return .standard
+        case .aggressive: return .aggressive
+        }
+      }
```
Comment (kylehickinson) [inline Sources/Preferences/GlobalPreferences.swift]: This and the `P3AAnswer` type should just go with the P3A answer, no need to define it in the pref

File: Sources/Preferences/GlobalPreferences.swift [hunk 2]
```diff
@@ -34,10 +34,82 @@ extension Preferences {
   }
   
   public final class Shields {
+    /// A 3 part option for shield levels varying in strength of blocking content
+    public enum ShieldLevel: String, CaseIterable {
```
Comment (kylehickinson) [inline Sources/Preferences/GlobalPreferences.swift]: Move this type to `BraveShields`, doesn't belong in prefs

Comment (cuba) [reply Sources/Preferences/GlobalPreferences.swift]: If I do this I will have to make the booleans public I'm trying to restrict access to this setting.

Comment (kylehickinson) [reply Sources/Preferences/GlobalPreferences.swift]: The booleans are already public

Comment (cuba) [reply Sources/Preferences/GlobalPreferences.swift]: I was hiding them (making the private) so nobody uses them directly but I can make them private again.

**Comment by cuba** [comment]
This PR is replaced with the following ones:
1. https://github.com/brave/brave-ios/pull/7354
2. https://github.com/brave/brave-ios/pull/7437 

The two will use swiftui for the advanced privacy settings but do not change the site specific settings. The site specific setting will be added later.

**Pull request closed without merging**